File: frontend/src/types/recommendation.ts
import type { Book } from "@/types/book";

export interface RecommendationItem {
  book: Book;
  match_score: number;
  reason: string;
  mood_tags: string[];
}

export interface RecommendNextResponse {
  recommendations: RecommendationItem[];
  unread_pool_size: number;
  criteria_summary: string;
}

File: backend/app/schemas/recommendation.py
from pydantic import BaseModel
from app.schemas.book import BookOut


class RecommendNextRequest(BaseModel):
    mood: str | None = None
    max_pages: int | None = None
    preferred_genre: str | None = None
    custom_prompt: str | None = None


class RecommendationItem(BaseModel):
    book: BookOut
    match_score: int
    reason: str
    mood_tags: list[str] = []


class RecommendNextResponse(BaseModel):
    recommendations: list[RecommendationItem]
    unread_pool_size: int
    criteria_summary: str
